text/0000-router-link-and-helpers.md

```diff
@@ -0,0 +1,161 @@
+
+    examples:
+
+      ```hbs
+      <button {{action (transition to='posts.edit' postId=post.id)}} />
+        Edit
+      </button>
+
+      <button {{action (transition to='login' replace=true)}} />
+        Login
+      </button>
+      ```
+
+
+
+ - `is-route-active`
+    this returns a boolean representing whether or not the current route matches the passed argument.
+
+    example:
+
+      ```hbs
+      <button
+        {{action (transition to='posts.edit' post=model.post)}}
+        class={{if (is-route-active 'posts.edit' post=model.post) 'selected'}} />
+      ```
+
+   this may need to support multiple invocation styles, for when there aren't parameters and the dev wants to be concise:
+
+   examples:
+
+     ```hbs
+     <button ... class={{if (is-route-active 'posts') 'selected'}} />
+      No model params provided
+     </button>
+     ```
+
+#### Phase 2
+
+*Add `Link` Component*
+
+`Link` could just be tagless component with the following template:
+
+```hbs
+{{!-- components/link/template.hbs --}}
+<a
+  {{action (transition @to models=@models)}}
+  class="{{if (is-route-active @to models=@models) 'active'}}">
+
+  {{yield}}
+
+</a>
+```
+
+Usage:
+```hbs
+<Link @to='posts.edit' @models={{post}}>Edit Post</Link>
+```
+
+where `@models` can be a `hash`, `array`, or just a single object or id.
+
+
+
+
+**Deprecation** `LinkTo` aka `link-to`
+
+The goal of `Link` and the route helpers is to provide a flexible way of routing from the template while providing a sample component with sensible defaults. This would achieve the exact same functionality as `LinkTo`, so `LinkTo` would no longer be needed and could eventually be removed.
+
+It's possible we could write a codemod to auto-convert everyone's non-angle-bracket invocation of `{{#link-to ...` to the new angle bracket component: `Link`
+
+## How we teach this
+
+We'll want to make it very clear that the traditional `LinkTo` technique of linking will still be available, but will eventually be deprecated.
+
+The documentation and guides would need to be incrementally upgraded to inform people about the new linking strategy -- but becaus the old way would still work, having some docs say `LinkTo` instead of `Link` wouldn't be the worst thing.
+
+
+## Drawbacks
+
+The biggest drawback is that all routing documentation would be out of date, and there is a lot of routing documentation and blog posts throughout the web.
+
+Without positional params, the alternative may need to use the `array` helper for the `@to` argument... which could almost make people wonder why `LinkTo` didn't get a named argument for the route / route params. With the `Link` component, the arguments would read more ergonomically -- `<Link @to=...` vs `<LinkTo @route=...`. The latter implies that things other than routes could be linked to via the `LinkTo` component, which could cause confusion about the intended usage.
+
+## Alternatives
+
+There are two alternatives:
+
+1. Do nothing:
```

Comment:
I'm thinking we should skip including this, because that is the alternative for every RFC, and the body of the RFC is an active argumentation against doing nothing.